### ISSUE TITLE
Memory leak after opening Cataclysm Ret Sim

### ISSUE THREAD

**Issue by BGannon2** [wowsims/cata#876]
**Describe the bug**
When you open the Ret sim and just let it sit on the sim without doing anything it will eventually just eat up all of the available RAM on your system.

v0.0.47 on Windows 11 using Firefox

**To Reproduce**
1. Download the sim
2. Open Task Manager (control + shift + escape)
3. Open and run the sim
4. Open the Ret Sim
5. Sort Task Manager by Memory and watch the sim eat ram

**Expected behavior**
The Sim runs with minimum RAM while idle.

**Sim Links and Screenshots**
![image](https://github.com/user-attachments/assets/49b4dc6c-0c8b-4264-9e4e-9091cfeab2e9)

**Desktop (please complete the following information):**
 - OS: Windows 11 
 - Browser: Firefox
 - Version: v0.0.47
![image](https://github.com/user-attachments/assets/fade8814-ab27-49a3-859f-99725da5b097)

**Comment by BGannon2** [comment]
@1337LutZ noted in discord that this is probably related to use of Firefox.

**Comment by FelixPflaum** [comment]
This has nothing to do with the browser, the wowsims.exe itself uses the memory.

**Comment by 1337LutZ** [comment]
> This has nothing to do with the browser, the wowsims.exe itself uses the memory.

Ahhhh yea, I misread it.

**Comment by 1337LutZ** [comment]
> @BGannon2 I also see this issue with Chrome. It started happening after I attempted to use the Batch feature once. Now, each time I load the Windows exe version of the Wowsims app, the memory leak is happening.

So even if you don't use the Batch sim it will leak? 🤔

**Comment by gilp456** [comment]
Yes, but I can't get it to reproduce cleanly. The first time it happened I imported my gear into the batch sim. The program soft hung and I was forced to close it via task manager. After that it always showed the memory leak symptom when I launched wowsim windows exe. I tried spinning up a windows virtual machine to repro but was unsuccessful so far. Will keep trying.

**Comment by Krokette** [comment]
Yeah same issue on chrome. I downloaded it just to be able to use the batch feature and it's almost impossible to use due to the huge memory leak.

**Comment by JaElvf** [comment]
This happens with [Arms Warrior](https://github.com/wowsims/cata/issues/1167) too

**Comment by 1337LutZ** [comment]
> Yes, but I can't get it to reproduce cleanly. The first time it happened I imported my gear into the batch sim. The program soft hung and I was forced to close it via task manager. After that it always showed the memory leak symptom when I launched wowsim windows exe. I tried spinning up a windows virtual machine to repro but was unsuccessful so far. Will keep trying.

You should be able to mitigate it by clearing the Batch sim options. That works for me at least on all the specs
